### PR TITLE
Generate DOT graphs from LLB states

### PR DESCRIPTION
**Pull request by NiR-**
This PR adds a new command to zbuild: `zbuild llbgraph`. It can be used to represent LLB graphs in DOT language (feh is used to open up the generated PNG file).:

```bash
$ zbuild debug-llb --json -f zbuild.yml --stage prod | zbuild llbgraph | dot /dev/stdin -o /dev/stdout -T png | feh -
```

This command can be used in a few contexts:

* For end-users, to get insights about what build steps zbuild generates ;
* For maintainers, to more easily review state files, even without deep knowledge about Buildkit ;
* For maintainers, to easily generate diagrams for each and every kind handler ;

Note that I had to change the way LLB ops are dumped into JSON to include their base64-encoded protobuf representation in binary format. This is the only way to easily unmarshal LLB ops back into protobuf structs. As such, all of the testdata files in JSON have been regenerated.

Here's an example for the new NodeJS handler:

nodejs-prod

And here's another example for the PHP handler:

php

### CODE REVIEW

File: cmd/zbuild/root.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ func main() {
 
 	zbuildCmd.AddCommand(newUpdateCmd())
 	zbuildCmd.AddCommand(newDebugLLBCmd())
+	zbuildCmd.AddCommand(newLlbgraphCmd())
```
Comment (nm2107) [inline cmd/zbuild/root.go]: `Llbgraph` => `LLBGraph` ? (as the above line)

File: pkg/llbgraph/llbgraph.go [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+		meta := def.Metadata[dgst]
+		if err := g.addOp(op, meta, dgst); err != nil {
+			return nil, err
+		}
+	}
+
+	return g, nil
+}
+
+func newDotGraph(opts GraphOpts) *DotGraph {
```
Comment (nm2107) [inline pkg/llbgraph/llbgraph.go]: :notbad:

File: pkg/llbgraph/testdata/nodejs-webserver-prod.dot [hunk 3]
```diff
@@ -0,0 +1,21 @@
+digraph llbgraph {
+"layer_sha256:06f33e5827a64a3d37fbbe5832c3dc6e525ad97f3f735d3b0525b402834ac61b" [label="sha256:06f33e5827a64a3d37fbbe5832c3dc6e525ad97f3f735d3b0525b402834ac61b",shape="box",style="",fillcolor=""]
```
Comment (nm2107) [inline pkg/llbgraph/testdata/nodejs-webserver-prod.dot]: this format seems pretty simple :)

Comment (NiR-) [reply pkg/llbgraph/testdata/nodejs-webserver-prod.dot]: Yep, the hardest part was to find a way to reconstruct the graph from JSON files :sweat_smile:

**Pull request merged**